packages/front-end/components/DataViz/DataVizFilter.tsx

```diff
@@ -0,0 +1,472 @@
+    if (value != null) {
+      uniqueValues.add(String(value));
+    }
+  });
+
+  return Array.from(uniqueValues);
+}
+
+const filterOptions = [
+  // Date filters
+  { value: "today", label: "Today", supportedTypes: ["date"] },
+  { value: "last7Days", label: "Last 7 Days", supportedTypes: ["date"] },
+  { value: "last30Days", label: "Last 30 Days", supportedTypes: ["date"] },
+  { value: "dateRange", label: "Custom Date Range", supportedTypes: ["date"] },
+
+  // Number filters
+  { value: "numberRange", label: "Custom Range", supportedTypes: ["number"] },
+  { value: "greaterThan", label: "Greater Than", supportedTypes: ["number"] },
+  {
+    value: "greaterThanOrEqualTo",
+    label: "Greater Than or Equal To",
+    supportedTypes: ["number"],
+  },
+  { value: "lessThan", label: "Less Than", supportedTypes: ["number"] },
+  {
+    value: "lessThanOrEqualTo",
+    label: "Less Than or Equal To",
+    supportedTypes: ["number"],
+  },
+  { value: "equalTo", label: "Equals", supportedTypes: ["number"] },
+
+  // String filters
+  { value: "includes", label: "Select Values", supportedTypes: ["string"] },
+  { value: "contains", label: "Text Search", supportedTypes: ["string"] },
+];
+
+export default function DataVizFilter({
+  filterIndex,
+  dataVizConfig,
+  onDataVizConfigChange,
+  rows,
+  columnFilterOptions,
+}: Props) {
+  const filters = dataVizConfig.filter || [];
+
+  const updateFilter = (newFilter: FilterConfiguration) => {
+    const newFilters = [...filters];
+    newFilters[filterIndex] = newFilter;
+    onDataVizConfigChange({
+      ...dataVizConfig,
+      filter: newFilters,
+    });
+  };
+
+  const updateFilterConfig = (
+    configUpdates: Record<string, string | number | string[] | undefined>,
+  ) => {
+    const currentFilter = filters[filterIndex];
+    const currentConfig = currentFilter?.config || {};
+
+    // Handle undefined values by deleting the key
+    const newConfig = { ...currentConfig };
+    Object.entries(configUpdates).forEach(([key, value]) => {
+      if (value === undefined) {
+        delete newConfig[key];
+      } else {
+        newConfig[key] = value;
+      }
+    });
+
+    // Create updated filter with new config, maintaining discriminated union structure
+    const updatedFilter: FilterConfiguration = {
+      ...currentFilter,
+      config: newConfig,
+    } as FilterConfiguration;
+
+    updateFilter(updatedFilter);
+  };
+
+  const removeFilter = () => {
+    const newFilters = [...filters];
+    newFilters.splice(filterIndex, 1);
+    onDataVizConfigChange({
+      ...dataVizConfig,
+      filter: newFilters,
+    });
+  };
+
+  const createDefaultFilterForType = (
+    column: string,
+    type: "string" | "number" | "date",
+  ): FilterConfiguration => {
+    if (type === "date") {
+      // Default to last 30 days for date filters
+      const today = new Date();
+      const thirtyDaysAgo = new Date(today);
+      thirtyDaysAgo.setDate(today.getDate() - 30);
+
+      return {
```

Comment:
Should this logic of the `default` state of a filter be merged with DataVizFilterPanel logic lines 185-214?